### ISSUE TITLE
add rayon support for PointCloudReader

### ISSUE THREAD

**Issue by omelette-watin** [cry-inc/e57#7]
Hello, 

thanks for this very useful crate! 
Would you consider implementing the [`ParallelIterator`](https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html) trait from rayon to `PointCloudReaderSimple`?
Currently i have to `.collect()` the reader before being able to use `.par_iter` which is kinda slow on big pointcloud.

Is that something you would consider? 
Thanks a lot

**Comment by cry-inc** [comment]
Have you tried to use `.par_bridge()` to create an parallel iterator from the existing one using a [ParallelBridge](https://docs.rs/rayon/latest/rayon/iter/trait.ParallelBridge.html)?

`let par_iter = reader.pointcloud_simple(&pointcloud).unwrap().par_bridge();`

But I think this would only speed up your code for very expensive per point operations. Otherwise the reading and parsing of the points from the E57 is still the bottleneck. Its currently slower than it should be and I will try to speed up reading in the future.

Thats also why I think it would be currently a bit weird to directly add support for the `ParallelIterator` trait.

**Comment by pnodet** [comment]
Hi @cry-inc 

We are currently working on a project where we needed to convert [e57 files to LAS](https://github.com/wildweb-io/e57-to-las-rs) and used your crate.

We tried to use `par_bridge()` and found that it was not worth it.

This issue can be closed now. Thanks for the work.

**Comment by cry-inc** [comment]
@pnodet & @omelette-watin: Thanks for the feedback and closing the issue. Your converter sounds like a useful thing!

Inspired by your project I added a very basic and minimalistic [E57-to-LAZ](https://github.com/cry-inc/e57/blob/master/tools/e57-to-laz/src/main.rs) example to my repo. Its mainly there as another example how to use the simple iterator interface.

I just released [0.8.0](https://crates.io/crates/e57/0.8.0), which comes (just like 0.7.0) with another perfomance improvement for reading E57 files. I also (once again) refactored the simple iteratior interface and the Point struct a bit, so it might cause some inconvenience when upgrading. Sorry for that! At least the project has now a proper [change log](https://github.com/cry-inc/e57/blob/master/CHANGELOG.md) :)
